Assignment/Program.c

```diff
@@ -4,11 +4,28 @@
 // Vinitha Perumal - GET
 // ------------------------------------------------------------------
 // Program.c
-// Program on main branch.
+// Program on A5 branch.
 // ------------------------------------------------------------------------------------------------
-#include <stdio.h>
 #include "Program.h"
 
-void PrintHelloWorld () {
-	printf ("Hello, World\n");
+void InsertionSort (int arr[], int n) {
+   int i, j, key;
+   for (i = 1; i < n; i++) {
+      key = arr[i], j = i - 1;
+      while (j >= 0 && arr[j] > key) arr[j + 1] = arr[j--];
+      arr[j + 1] = key;
+   }
+}
+
+int BinarySearch (int arr[], int n, int key) {
+   int low = 0, high = n - 1, res = -1, mid;
+   while (low <= high) {
+      mid = (low + high) / 2;
+      if (arr[mid] == key) {
+         res = mid;

```

Comment:
This is not Binary Search as it will continue to look for the element even after it has found it. It will in fact return the last occurrence of the value.